client/src/util/metrics/processVariables.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership.
+ *
+ * Camunda licenses this file to you under the MIT; you may not use this file
+ * except in compliance with the MIT License.
+ */
+
+import { getProcessVariables } from '@bpmn-io/extract-process-variables';
+import { parse } from './helpers';
+
+export async function extractProcessVariables(file) {
+  const { contents } = file;
+  const processVariables = [];
+  const definitions = await parse(contents);
+  const rootElements = definitions.get('rootElements');
+
+  rootElements.forEach((element) => processVariables.push(...getProcessVariables(element)));
```

Comment:
I'm personally not a big fan of these single-liners doing multiple stuff. But that's maybe a matter of taste.